docs/configuration/module-config/serial.mdx

```diff
@@ -294,3 +294,170 @@ previousState = currentState; // update the previous state
+
+      if (buttonState == LOW) { // if the button pin is still LOW after debounce delay, that means the detected change was a real press, so we send the relay toggle command. If not, then we just assume the state change was noise and do nothing.
+
+        sendRelayToggleCommand();
+
+      }
+
+    }
+
+  }
+
+}
+
+void sendRelayToggleCommand() {
+
+  Serial.println(relayToggleCode);
+
+}
+
+```
+
+#### Arduino Mini Pro / relay Wiring
+
+![image](/img/modules/Serial/relay.png)
+
+- T-BEAM RX PIN 13 to TX PIN on the ARDUINO MINI
+- T-BEAM TX PIN 14 to RX PINon the ARDUINO MINI
+- T-BEAM PIN 3.3V to 3.3V PIN on the ARDUINO MINI
+- T-BEAM PIN GND to GND PIN on the ARDUINO MINI
+- ARDUINO MINI PIN 2 to SIGNAL PIN on the RELAY
+- ARDUINO MINI PIN GND to the GND on the RELAY
+- ARDUINO MINI PIN VCC to the VCC on the RELAY
+
+#### Arduino Mini Pro /relay Code
+
+```cpp
+//define sketch constants (gpios, baud rate and other constants)
+#define RELAY_PIN 2
+#define BAUD_RATE 9600
+
+int currentRelayState;
+const char* relayToggleCode = "relay1toggle";
+const char* relayOnCode = "on";
+const char* relayOffCode = "off";
+const unsigned int MAX_INPUT = 30;
+
+void setup() {
+  Serial.begin(BAUD_RATE);
+  pinMode(RELAY_PIN, OUTPUT);
+  digitalWrite(RELAY_PIN, LOW);
+  currentRelayState = LOW;
+}
+
+void loop() {
+  while (Serial.available() > 0) {
+    processIncomingByte(Serial.read());
+  }
+}
+
+void processIncomingByte(const byte inByte) {
+  static char input_line[MAX_INPUT];
+  static unsigned int input_pos = 0;
+
+  switch (inByte) {
+    case '\n':
+      input_line[input_pos] = 0;
+      checkStringForPasscode(input_line);
+      input_pos = 0;
+      break;
+    case '\r':
+      break;
+    default:
+      if (input_pos < (MAX_INPUT - 1))
+        input_line[input_pos++] = inByte;
+      break;
+  }
+}
+
+void checkStringForPasscode(const char* data) {
+  String inputString = data;
+
+  if (inputString.indexOf(relayToggleCode) != -1) {
+    if (currentRelayState == LOW) {
+      digitalWrite(RELAY_PIN, HIGH);
+      currentRelayState = HIGH;
+      const char* confirmationMessage = "Relay 1 has received the command and is now ON";
+      Serial.println(confirmationMessage);
+    } else {
+      digitalWrite(RELAY_PIN, LOW);
+      currentRelayState = LOW;
+      const char* confirmationMessage = "Relay 1 has received the command and is now OFF";
+      Serial.println(confirmationMessage);
+    }
+  } else if (inputString.indexOf(relayOnCode) != -1) {
+    digitalWrite(RELAY_PIN, HIGH);
+    currentRelayState = HIGH;
+    const char* confirmationMessage = "Relay 1 has received the command and is now ON";
+    Serial.println(confirmationMessage);
+  } else if (inputString.indexOf(relayOffCode) != -1) {
+    digitalWrite(RELAY_PIN, LOW);
+    currentRelayState = LOW;
+    const char* confirmationMessage = "Relay 1 has received the command and is now OFF";
+    Serial.println(confirmationMessage);
+  }
+}
+```
```

Comment:
This sketch would benefit from some comments to help users understand how it functions.

Reply:
Thanks for the help @pdxlocations